### ISSUE TITLE
Missing TMB in Kids_First_Biospecimen_ID in the PBTA - rerun tmb-calculation module

### ISSUE THREAD

**Issue by AntoniaChroni** [PediatricOpenTargets/OpenPBTA-analysis#430]
#### What data file(s) does this issue pertain to?

This is related to the `snv-mutation-tmb-coding.tsv` file.

#### What release are you using?

v12

#### Put your question or report your issue here.

There is missing TMB information for some `Kids_First_Biospecimen_ID` in the PBTA cohort. This will result in cases with no matched longitudinal samples (`tumor_descriptor`) per `Kids_First_Participant_ID`. 

I compared the `tmb` to the `pbta` file, and filter out any missing data (see attached file named as `tmb_missing_bs.tsv`).
Input files are located at `analyses/tmb-vaf-longitudinal/input/snv-mutation-tmb-coding.tsv` and `analyses/sample-distribution-analysis/results/pbta.tsv`, respectively.

Can we find out why this is happening?

**Comment by jharenza** [comment]
@AntoniaChroni can you add the list of missing bs ids to this ticket please? Also, I will transfer this over to the OPC repo as it deals with the underlying data there. Thanks!

**Comment by AntoniaChroni** [comment]
[tmb_missing_bs.xlsx](https://github.com/d3b-center/OpenPedCan-analysis/files/12388882/tmb_missing_bs.xlsx)

**Comment by jharenza** [comment]
Quick comparison of the relevant files suggests this module was not rerun quite right for v12, but all samples are in the v12 `biospecimen_id_to_bed_map.tsv` file, meaning the TMB should be able to be generated.
```r
> tmb_release <- read_tsv("~/Documents/GitHub/ped-ot/OpenPedCan-analysis/data/v12/snv-mutation-tmb-coding.tsv")
Rows: 2764 Columns: 5                                                                                                                                                 
── Column specification ───────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): Tumor_Sample_Barcode, experimental_strategy
dbl (3): mutation_count, region_size, tmb

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
> tmb_repo <- read_tsv("~/Documents/GitHub/ped-ot/OpenPedCan-analysis/analyses/tmb-calculation/results/snv-mutation-tmb-coding.tsv")
Rows: 2764 Columns: 5                                                                                                                                                 
── Column specification ───────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): Tumor_Sample_Barcode, experimental_strategy
dbl (3): mutation_count, region_size, tmb

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
> tmb_missing <- readxl::read_excel("~/Downloads/tmb_missing_bs.xlsx")
                                                                                                                                 
> bs_map <- read_tsv("~/Documents/GitHub/ped-ot/OpenPedCan-analysis/data/v12/biospecimen_id_to_bed_map.tsv")
Rows: 4414 Columns: 2                                                                                                                                                 
── Column specification ───────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): Kids_First_Biospecimen_ID, bed_to_use

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
> length(setdiff(tmb_missing$Kids_First_Biospecimen_ID, tmb_release$Tumor_Sample_Barcode))
[1] 1387
> length(setdiff(tmb_missing$Kids_First_Biospecimen_ID, tmb_repo$Tumor_Sample_Barcode))
[1] 1387
> length(setdiff(bs_map$Kids_First_Biospecimen_ID, tmb_release$Tumor_Sample_Barcode))
[1] 1650
> length(setdiff(tmb_missing$Kids_First_Biospecimen_ID, bs_map$Kids_First_Biospecimen_ID))
[1] 0
```

Given that all of the data is present for v12, can you simply do a rerun of [this](https://github.com/d3b-center/OpenPedCan-analysis/tree/dev/analyses/tmb-calculation) module on a branch and submit a PR? That should get you the complete file.

**Comment by AntoniaChroni** [comment]
I edited `input_dir `to `data_dir` in the shell file. However, this gave the following error:

```
Error: '/home/rstudio/OpenPedCan-analysis/data/ashion_confidential_exome_v2_targets_hg38_paded100.bed' does not exist.
Execution halted
```

Bear in mind that in the previous version, it was `ashion_exome_v2_targets_hg38_paded100.bed` in the input file.

This happens because there is no `ashion_confidential_exome_v2_targets_hg38_paded100.bed` file in the v12 data_dir.

Is the `ashion_confidential_exome_v2_targets_hg38_paded100.bed` file supposed to be in the v12 or could it be that indeed `ashion_exome_v2_targets_hg38_paded100.bed` is needed to run the `tmb-calculation` module?

@jharenza

**Comment by jharenza** [comment]
Ahhh - yes the name of that bed file was changed. @ewafula we are trying to understand a few things with this module. 1) it seems the biospecimen id to bed mapping file being used here is used from input not data and wasn't updated in input to match the data release file and 2) the rename of the bed file mentioned above wasn't captured across the code and files fully. Is that a full assessment, @AntoniaChroni ?

**Comment by ewafula** [comment]
@jharenza and @AntoniaChroni  `ashion_exome_v2_targets_hg38_padded100.bed` is in the v12 s3 bucket.
https://d3b-openaccess-us-east-1-prd-pbta.s3.amazonaws.com/open-targets/testing_v12/ashion_exome_v2_targets_hg38_padded100.bed

The bash script code line below needs to be updated to read the bed mapping file from the data directory, not the module input directory. The module is currently reading an old mapping file from previous releases.  @AntoniaChroni, I also see a typo in the bed file name in your comment above: `ashion_exome_v2_targets_hg38_paded100.bed` instead of `ashion_exome_v2_targets_hg38_padded100.bed` (padded is misspelled). Suggests that your `biospecimen_id_to_bed_map.tsv` is not  even the one currently in module [input_dir](https://raw.githubusercontent.com/d3b-center/OpenPedCan-analysis/dev/analyses/tmb-calculation/input/biospecimen_id_to_bed_map.tsv)

Change line 27 to this:
```
# sample to BED mapping file
mapping_file="${data_dir}/biospecimen_id_to_bed_map.tsv"
```
https://github.com/d3b-center/OpenPedCan-analysis/blob/e1197325a72cc93d93b1acf7bdb42ef70ec6ec64/analyses/tmb-calculation/run_tmb_calculation.sh#L25-L28

**Comment by AntoniaChroni** [comment]
@ewafula

I modified the shell to read the bed mapping file from `data_dir`. However, that bed mapping file tries to call `ashion_confidential_exome_v2_targets_hg38_paded100.bed` file instead of  `ashion_exome_v2_targets_hg38_paded100.bed` (the latter is in the `data_dir`). @jharenza says that the filename was edited at some point, so we need to update the filename in the bed mapping file in `data_dir` to reflect that.

As for the misspelling, I confirm that the one in the `input_dir `has the correct filename and spelling. 

I re-run the shell after calling the bed file from `data_dir` (that one contains the incorrect filename and it's misspelled) and that is when I got the error.

I wanted to re-run the module because the one currently contains less number of bs_id with TMB than it should in the PBTA cohort (`snv-mutation-tmb-coding.tsv`).

**Comment by ewafula** [comment]
@AntoniaChroni , `ashion_confidential_exome_v2_targets_hg38_paded100.bed`  is not currently in the v12/  data release folder on s3 bucket.  Likely the reason why the module is failing. 

@zhangb1 uploaded all the bed files associated with v12 samples required for running the TMB module to the s3 bucket and provided me with the list of BS_IDs to BED_File mapping (`v12/biospecimen_id_to_bed_map.tsv`). He must have inadvertently forgotten to upload `ashion_confidential_exome_v2_targets_hg38_paded100.bed` bed file. 

@zhangb1, can you please upload the be `ashion_confidential_exome_v2_targets_hg38_paded100.bed` bed file to `v12 s3 folder` associated with the following BS_IDs.  It seems you did not upload it.  @AntoniaChroni can then download to her local OPC data directory
```
BS_02QV7ZWP	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_05CHNS17	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_0TKQP4XJ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_13DJ6WHW	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_1B00Q25Y	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_1SDWBXX8	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_35DRDAFV	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_53JGYCCZ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_5XC635HW	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_5YTKXXBY	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_DD69EB5S	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_F6YJDSXZ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_HF1TR0QN	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_HGKDT5AJ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_PCVV9GJG	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_W60PB1DS	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_YGDHJ68W	ashion_confidential_exome_v2_targets_hg38_paded100.bed
```
@AntoniaChroni, can you check if there is any bed file listed in the `biospecimen_id_to_bed_map.tsv` (https://d3b-openaccess-us-east-1-prd-pbta.s3.amazonaws.com/open-targets/testing_v12/biospecimen_id_to_bed_map.tsv) in the column `bed_to_use` missing the OPC v12 data download?  @zhangb1 should the upload to s3 if any. 

Cc @jharenza

**Comment by jharenza** [comment]
@ewafula I think we changed the name of that bed file not to have "confidential" in it, but missed this in the mapping file. We do _NOT_ want to change any files after the data is released @zhangb1 - nothing to do for v12. Will update in v13 per #434

**Comment by ewafula** [comment]
@jharenza, so what file change?  `ashion_confidential_exome_v2_targets_hg38_paded100.bed` currently missing on s3 is listed `v12/biospecimen_id_to_bed_map.tsv`.  Did the `v12/biospecimen_id_to_bed_map.tsv` mapping updated with the above samples and BED file as well?  The BS_IDs and BED file  listed above are from v`12/biospecimen_id_to_bed_map.tsv`

**Comment by jharenza** [comment]
`ashion_confidential_exome_v2_targets_hg38_padded100.bed` became `ashion_exome_v2_targets_hg38_padded100.bed`. It looks like the `v12/biospecimen_id_to_bed_map.tsv` or `input/biospecimen_id_to_bed_map.tsv` are missing all v12 samples and also have the typo, so double whammy!

**Comment by ewafula** [comment]
Both bed files that are supposed to replace each other are listed in the `v12/biospecimen_id_to_bed_map.tsv` on s3 and associated with different sets of samples!!
```
[wafulae@reslndbhisbox03 TEMP]$ grep 'ashion_*' /mnt/isilon/opentargets/s3-opentargets/s3-data/v12/biospecimen_id_to_bed_map.tsv 
BS_02QV7ZWP	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_05CHNS17	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_0TKQP4XJ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_13DJ6WHW	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_1B00Q25Y	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_1SDWBXX8	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_35DRDAFV	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_53JGYCCZ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_5XC635HW	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_5YTKXXBY	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_DD69EB5S	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_F6YJDSXZ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_HF1TR0QN	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_HGKDT5AJ	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_PCVV9GJG	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_W60PB1DS	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_YGDHJ68W	ashion_confidential_exome_v2_targets_hg38_paded100.bed
BS_3DV5FVPQ	ashion_exome_v2_targets_hg38_padded100.bed
BS_AG4BP2PM	ashion_exome_v2_targets_hg38_padded100.bed
BS_AMKQKHJ6	ashion_exome_v2_targets_hg38_padded100.bed
BS_AN55105E	ashion_exome_v2_targets_hg38_padded100.bed
BS_BTDQZ60Q	ashion_exome_v2_targets_hg38_padded100.bed
BS_C5M87D60	ashion_exome_v2_targets_hg38_padded100.bed
BS_GBXD55F1	ashion_exome_v2_targets_hg38_padded100.bed
BS_GJ2KS7BT	ashion_exome_v2_targets_hg38_padded100.bed
BS_GQFPB8F3	ashion_exome_v2_targets_hg38_padded100.bed
BS_HSXARQ1K	ashion_exome_v2_targets_hg38_padded100.bed
BS_PZ5WX9TX	ashion_exome_v2_targets_hg38_padded100.bed
BS_QJHY513X	ashion_exome_v2_targets_hg38_padded100.bed
BS_S3DSBR9Y	ashion_exome_v2_targets_hg38_padded100.bed
BS_ST7KGV85	ashion_exome_v2_targets_hg38_padded100.bed
BS_VS3R7XVV	ashion_exome_v2_targets_hg38_padded100.bed
BS_XHRV3SAK	ashion_exome_v2_targets_hg38_padded100.bed
BS_ZV7AC5R8	ashion_exome_v2_targets_hg38_padded100.bed
```

**Comment by jharenza** [comment]
I know 🤦‍♀️ !

**Comment by jharenza** [comment]
closed with #435
